lib/models/villager_model.dart

```diff
@@ -0,0 +1,42 @@
+class villager {
+  String? name;
+  String? imageUrl;
+  String? species;
+  String? personality;
+  String? gender;
+  String? birthdayMonth;
+  String? birthdayDay;
+
+  villager({
+    this.name,
+    this.imageUrl,
+    this.species,
+    this.personality,
+    this.gender,
+    this.birthdayMonth,
+    this.birthdayDay,
+  });
+
+  villager.fromJson(Map<String, dynamic> json) {
+    name = json['name'];
+    imageUrl = json['image_url'];
+    species = json['species'];
+    personality = json['personality'];
+    gender = json['gender'];
+    birthdayMonth = json['birthday_month'];
+    birthdayDay = json['birthday_day'];
```

Comment:
I think it's better to typecast these and make sure the data types are coming as expected.